timesketch/api/v1/resources/sigma.py

```diff
@@ -0,0 +1,220 @@
+                    print(rule_filename)
+                    # if a sub dir is found, append it to be scanned for rules
+                    if os.path.isdir(os.path.join(rules_path, rule_filename)):
+                        logger.error(
+                            'this is a directory, skipping: {0:s}'.format(
+                                rule_filename))
+                        continue
+
+                    tag_name, _ = rule_filename.rsplit('.')
+                    rule_file_path = os.path.join(dirpath, rule_filename)
+                    rule_file_path = os.path.abspath(rule_file_path)
+
+                    with codecs.open(rule_file_path, 'r', encoding='utf-8',errors='replace') as rule_file:
+                                try:
+                                    rule_file_content = rule_file.read()
+                                    rule_yaml_data = yaml.safe_load(rule_file_content)
+                                    sigma_rules.append(rule_yaml_data)
+                                except NotImplementedError as exception:
+                                    logger.error(
+                                        'Error generating rule in file {0:s}: {1!s}'
+                                        .format(rule_file_path, exception))
+                                    continue
+        
+        # TODO: is that actually needed?
+        meta = {'current_user': current_user.username}
+        return jsonify({'objects': sigma_rules, 'meta': meta})
+
+
+class SigmaResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule."""
+
+    def __init__(self):
+    
+        super(SigmaResource, self).__init__()
+        
+
+    @login_required
+    def get(self, rule_uuid):
+        """Handles GET request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+
+        logger.info(rule_uuid)
+
+        # TODO: that should be part of the Timesketch config.
+        try:
+            _CONFIG_FILE = current_app.config['SIGMA_CONFIG']
+        except KeyError:
+            logger.error("SIGMA_CONFIG not found in config file")
+            _CONFIG_FILE = '../../../../data/sigma_config.yaml'
+
+        # Path to the directory containing the Sigma Rules to run, relative to
+        # this file.
+        # TODO check if get is better for getting the key errors
+        try:
+            _RULES_PATH = current_app.config['SIGMA_RULES_FOLDER']
+        except KeyError:
+            logger.error("SIGMA_RULES_FOLDER not found in config file")
+            _RULES_PATH = '../../../../data/sigma/rules/'
+
+        sigma_config_path = os.path.join(os.path.dirname(__file__), _CONFIG_FILE)
+
+        with open(sigma_config_path, 'r') as sigma_config_file:
+            sigma_config_file = sigma_config_file.read()
+        sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+    
+        sigma_backend = sigma_elasticsearch.ElasticsearchQuerystringBackend(sigma_config, {})
```

Comment:
will stop with the linter, but line too long

and please make sure to fix all linter issues